src/sstruct_mv/sstruct_matrix.c

```diff
@@ -875,15 +875,64 @@ hypre_SStructUMatrixInitialize( hypre_SStructMatrix  *matrix,
       nrows    = hypre_SStructGridLocalSize(grid);
    }
 
-   /* Set row sizes */
 #if defined(HYPRE_USING_GPU)
    if (exec == HYPRE_EXEC_DEVICE)
    {
+      /* Set max_size (row_sizes not needed for GPU execution) */
       row_sizes = NULL;
+      for (part = 0; part < nparts; part++)
```

Comment:
Hi @waynemitchell .  Not sure if you are ready yet with this PR, but I wanted to give you a quick reaction.  There is a lot of code duplication here, so if it's possible to list the replicated code once outside of the if/else construct, that would be preferable to me.  That is, minimize the specialized GPU code as much as possible.  Thanks!